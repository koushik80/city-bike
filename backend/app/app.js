// external imports
const express = require("express");
const app = express();
const dotenv = require("dotenv");
const cors = require("cors");

// dot environment config
if (process.env.NODE_ENV !== "PRODUCTION") {
    dotenv.config({ path: 'backend/config/config.env' });
}

// parsing json
app.use(express.json());

// used cors to allow cross-origin access with frontend
app.use(cors());

// imported routers (internal imports)
const journey = require("../routes/journey.routes");
const station = require("../routes/station.routes");
const { errorMiddleware } = require("../middleware/errorMiddleware");

// middleware
app.use("/api/journey", journey);
app.use("/api/station", station);


app.get('/', (req, res) => {
    res.status(200).json({
        message: "This is JOURNEY REST API"
    });
});

// ERROR MIDDLEWARE
app.use(errorMiddleware)

module.exports = app;