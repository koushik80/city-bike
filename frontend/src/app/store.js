// https://redux-toolkit.js.org/rtk-query/overview

import { configureStore } from '@reduxjs/toolkit'
import apiSlice from '../features/api/apiSlice'
import filterReducer from '../features/filter/filterSlice'

export const store = configureStore({
  reducer: {
    [apiSlice.reducerPath]: apiSlice.reducer,
    filter: filterReducer
  },
  devTools: process.env.NODE_ENV !== 'production',
  middleware: (getDefaultMiddlewares) => getDefaultMiddlewares().concat(apiSlice.middleware)
})