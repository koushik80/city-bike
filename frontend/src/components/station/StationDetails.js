import { Container } from '@mui/material';
import React from 'react';
import { useParams } from 'react-router-dom';
import { useGetStationDetailsQuery } from '../../features/station/stationApi';
import './StationDetails.scss';

const StationDetails = () => {

    const { id } = useParams();
    const { data } = useGetStationDetailsQuery(id);
    console.log(data);

    return (
        <Container>
            <div className="station-details-page">
                <h1>id: {id}</h1>
            </div>
        </Container>
    );
};

export default StationDetails;