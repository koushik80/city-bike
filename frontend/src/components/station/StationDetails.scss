.station-details-page {
  padding: 1rem 2rem;
  margin: 2rem 0rem;

  &--info {
    font-size: 18px;

    li {
      margin-top: 8px;
      font-weight: 500;
    }
  }

  &--info-2 {
    font-size: 18px;

    li {
      margin-top: 8px;
      font-weight: 400 !important;
    }
  }
}