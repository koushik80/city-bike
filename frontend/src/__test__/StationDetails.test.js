import { shallow } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { store } from '../app/store';
import StationDetails from '../components/station/StationDetails';

describe("station details page test suite", () => {

    it("should render station details", () => {
        const wrapper = renderer.create(<Provider store={store}><StationDetails /></Provider>).toJSON();

        expect(wrapper).toMatchSnapshot();
    });

    it("Station details function testing", () => {
        const wrapper = shallow(<Provider store={store}><StationDetails /></Provider>);

        let getMostPopularReturnStation = wrapper.instance();

        console.log("MostPopularReturnStation", getMostPopularReturnStation)
    });
});