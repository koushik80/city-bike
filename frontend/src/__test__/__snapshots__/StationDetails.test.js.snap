// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`station details page test suite should render station details 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root station-details-page css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
    >
      <div
        className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
      >
        <span
          className="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
        >
          Station details
        </span>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-480o17-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 MuiGrid-grid-md-6 MuiGrid-grid-lg-6 css-wkfkmr-MuiGrid-root"
      >
        <span
          className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1kzild6-MuiSkeleton-root"
          style={
            Object {
              "height": undefined,
              "width": undefined,
            }
          }
        />
        <span
          className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1kzild6-MuiSkeleton-root"
          style={
            Object {
              "height": undefined,
              "width": undefined,
            }
          }
        />
        <span
          className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1kzild6-MuiSkeleton-root"
          style={
            Object {
              "height": undefined,
              "width": undefined,
            }
          }
        />
        <span
          className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1kzild6-MuiSkeleton-root"
          style={
            Object {
              "height": undefined,
              "width": undefined,
            }
          }
        />
        <span
          className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1kzild6-MuiSkeleton-root"
          style={
            Object {
              "height": undefined,
              "width": undefined,
            }
          }
        />
        <span
          className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1kzild6-MuiSkeleton-root"
          style={
            Object {
              "height": undefined,
              "width": undefined,
            }
          }
        />
        <span
          className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1kzild6-MuiSkeleton-root"
          style={
            Object {
              "height": undefined,
              "width": undefined,
            }
          }
        />
        <span
          className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1kzild6-MuiSkeleton-root"
          style={
            Object {
              "height": undefined,
              "width": undefined,
            }
          }
        />
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 MuiGrid-grid-md-6 MuiGrid-grid-lg-6 css-kdp3n0-MuiGrid-root"
      >
        <span
          className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1kzild6-MuiSkeleton-root"
          style={
            Object {
              "height": undefined,
              "width": undefined,
            }
          }
        />
        <span
          className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1kzild6-MuiSkeleton-root"
          style={
            Object {
              "height": undefined,
              "width": undefined,
            }
          }
        />
        <span
          className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1kzild6-MuiSkeleton-root"
          style={
            Object {
              "height": undefined,
              "width": undefined,
            }
          }
        />
        <span
          className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1kzild6-MuiSkeleton-root"
          style={
            Object {
              "height": undefined,
              "width": undefined,
            }
          }
        />
        <span
          className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1kzild6-MuiSkeleton-root"
          style={
            Object {
              "height": undefined,
              "width": undefined,
            }
          }
        />
        <span
          className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1kzild6-MuiSkeleton-root"
          style={
            Object {
              "height": undefined,
              "width": undefined,
            }
          }
        />
        <span
          className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1kzild6-MuiSkeleton-root"
          style={
            Object {
              "height": undefined,
              "width": undefined,
            }
          }
        />
        <span
          className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1kzild6-MuiSkeleton-root"
          style={
            Object {
              "height": undefined,
              "width": undefined,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
